crates/js-sys/src/lib.rs

```diff
@@ -3522,6 +3522,22 @@ impl JsString {
             None
         }
     }
+
+    /// Returns whether this string is a valid UTF-16 string.
+    ///
+    /// This is useful for learning whether `as_string()` will return a lossless
+    /// representation of the JS string. If this string contains lone surrogates
+    /// then `as_string()` will succeed but it will be a lossy representation of
+    /// the JS string because lone surrogates will become replacement
+    /// characters.
+    ///
+    /// If this function returns `false` then to get a lossless representation
+    /// of the string you'll need to manually use `char_code_at` accessor to
+    /// access the raw code points.
+    pub fn is_valid_utf16(&self) -> bool {
```

Comment:
Since we already have the code for it, maybe we should extract this into a method? Something like this...

```rust
pub fn iter(&self) -> impl Iterator<Item = u16>
```

Now it's quite easy to convert to `Vec<u16>` if the user wishes to do so.